deepmd/pt/utils/dataloader.py

```diff
@@ -139,8 +139,9 @@ def construct_dataset(system):
             self.total_batch += len(system_dataloader)
         # Initialize iterator instances for DataLoader
         self.iters = []
-        for item in self.dataloaders:
-            self.iters.append(iter(item))
+        with torch.device("cpu"):
+            for item in self.dataloaders:
+                self.iters.append(iter(item))
```

Comment:
Does this device setting matter here?

Reply:
https://github.com/pytorch/pytorch/blob/c28a43988e7f3c630e5c50717c146c2cd936eabf/torch/utils/data/dataloader.py#L602

Reply:
got it